rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -371,6 +377,10 @@ func (r *Layer2Relayer) ProcessPendingBatches() {
 		// send transaction
 		txID := batch.Hash + "-commit"
 		fallbackGasLimit := uint64(float64(batch.TotalL1CommitGas) * r.cfg.L1CommitGasLimitMultiplier)
+		if types.RollupStatus(batch.RollupStatus) == types.RollupCommitFailed {
+			// use estimateGas if this batch has been committed failed.
```

Comment:
Add a `Warn` log here.

Reply:
added in [74f93df](https://github.com/scroll-tech/scroll/pull/981/commits/74f93dfa41230e511f16bdca580ea5e43e008b45)